### PR TITLE
Feature/mobile drag drop quiz

### PR DESCRIPTION
**Pull request by lovaiible**
### Checklist
- [x] I run `yarn run webpack:build:main`: the project builds without errors.
- [x] I run `yarn lint`: the project builds without code style warnings.
- [x] ~~I updated the documentation and models.~~
- [x] I tested the changes and all related features on the test server https://artemistest.ase.in.tum.de.
- [x] ~~I added (end-to-end) test cases for the new functionality.~~

### Motivation and Context
Enable drag and drop quizzes on mobile devices and optimize the display of diagrams and drag and drop elements for broad desktop monitors.

### Description
The drag and drop functionality for quiz exercises does not work on mobile devices.
The drag and drop elements cannot be seen without scrolling if the desktop monitor is too broad.

### Steps for Testing
1. Create drag and drop quiz exercise
2. Navigate to exercise on mobile device
3. Try to solve the quiz by moving the drag and drop elements to the right drop locations

### Screenshots
image
video
video2

### CODE REVIEW

File: src/main/webapp/i18n/de/dragAndDropQuestion.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
             "addMappingsInstructions": "Ziehen Sie Drag Items auf Drop Locations um korrekte Zuordnungen zu definieren.",
             "addDragItemPicture": "Drag Item mit Bild Hinzufügen",
             "addDragItemText": "Drag Item mit Text Hinzufügen",
-            "studentInstructions": "Drag & Drop: Platzieren Sie die passenden Elemente unten auf die zugehörigen Flächen oben.",
+            "studentInstructions": "Drag & Drop: Platzieren Sie die passenden Elemente unten auf die zugehörigen Flächen des Diagrams.",
```
Comment (krusche) [inline src/main/webapp/i18n/de/dragAndDropQuestion.json]: It does not have to be a diagram. Can we reformulate it?

File: src/main/webapp/i18n/en/dragAndDropQuestion.json [hunk 2]
```diff
@@ -30,7 +30,7 @@
             "addMappingsInstructions": "Drag items onto drop locations to define correct solutions.",
             "addDragItemPicture": "Add Picture Drag Item",
             "addDragItemText": "Add Text Drag Item",
-            "studentInstructions": "Drag & Drop: Place the suitable items below on the correct areas above.",
+            "studentInstructions": "Drag & Drop: Place the suitable items below on the correct areas in the displayed diagram.",
```
Comment (krusche) [inline src/main/webapp/i18n/en/dragAndDropQuestion.json]: It does not have to be a diagram. Can we reformulate it?

**Comment by krusche** [comment]
Great! Could you provide some screenshots how this looks?

**Comment by lovaiible** [comment]
> Great! Could you provide some screenshots how this looks?

@krusche  I've added the screenshots :) However, I'm still facing the problem that the translucent ghost images of pictures are not visible when dragging the element (z.B. when dragging the arrow). I'll need more time to look into that.

**Comment by lovaiible** [comment]
> > Great! Could you provide some screenshots how this looks?
> 
> @krusche I've added the screenshots :) However, I'm still facing the problem that the translucent ghost images of pictures are not visible when dragging the element (z.B. when dragging the arrow). I'll need more time to look into that.

@krusche the "preview" images when dragging pictures works in the mobile simulator but not on the real device. I could only test it with iOS devices but not with Android phones.

**Comment by krusche** [comment]
What is the status here? I have in mind that there was still some progress regarding the "ghost images". Is it finished? Can you please quickly try it out with the recent changes with angular8? Thanks!

**Comment by lovaiible** [comment]
@krusche it's working and I have also tested it with the new angular update

File: src/main/webapp/app/components/util/secured-image.component.ts [hunk 3]
```diff
@@ -56,6 +63,7 @@ export class SecuredImageComponent implements OnChanges {
     }
 
     private loadImage(url: string): Observable<any> {
+        console.log('drag and drop: ', this.dragAndDrop);
```
Comment (krusche) [inline src/main/webapp/app/components/util/secured-image.component.ts]: Can we remove this line for production?

File: src/main/webapp/content/scss/quiz.scss [hunk 4]
```diff
@@ -137,9 +137,9 @@
     right: 0;
     z-index: 10;
 
-    @media (min-width: 320px) and (max-width: 480px) {
+    /*@media (min-width: 320px) and (max-width: 480px) {
```
Comment (krusche) [inline src/main/webapp/content/scss/quiz.scss]: why do you comment this out? Can we remove the code?

File: src/main/webapp/content/scss/quiz.scss [hunk 5]
```diff
@@ -214,9 +214,9 @@
 .quiz-footer {
     height: 100px;
 
-    @media (min-width: 320px) and (max-width: 480px) {
+    /*@media (min-width: 320px) and (max-width: 480px) {
```
Comment (krusche) [inline src/main/webapp/content/scss/quiz.scss]: why do you comment this out? Can we remove the code?

File: src/main/webapp/i18n/de/dragAndDropQuestion.json [hunk 6]
```diff
@@ -30,7 +30,7 @@
             "addMappingsInstructions": "Ziehen Sie Drag Items auf Drop Locations um korrekte Zuordnungen zu definieren.",
             "addDragItemPicture": "Drag Item mit Bild Hinzufügen",
             "addDragItemText": "Drag Item mit Text Hinzufügen",
-            "studentInstructions": "Drag & Drop: Platzieren Sie die passenden Elemente unten auf die zugehörigen Flächen oben.",
+            "studentInstructions": "Drag & Drop: Platzieren Sie die passenden Elemente unten auf die zugehörigen Flächen.",
```
Comment (krusche) [inline src/main/webapp/i18n/de/dragAndDropQuestion.json]:
```suggestion
            "studentInstructions": "Drag & Drop: Platziere die passenden Elemente auf die zugehörigen Flächen.",
```

File: src/main/webapp/i18n/en/dragAndDropQuestion.json [hunk 7]
```diff
@@ -30,7 +30,7 @@
             "addMappingsInstructions": "Drag items onto drop locations to define correct solutions.",
             "addDragItemPicture": "Add Picture Drag Item",
             "addDragItemText": "Add Text Drag Item",
-            "studentInstructions": "Drag & Drop: Place the suitable items below on the correct areas above.",
+            "studentInstructions": "Drag & Drop: Place the suitable items below on the correct areas.",
```
Comment (krusche) [inline src/main/webapp/i18n/en/dragAndDropQuestion.json]:
```suggestion
            "studentInstructions": "Drag & Drop: Place the suitable items on the correct areas.",
```

**Comment by lovaiible** [comment]
@krusche I have fixed the mentioned issues

**Comment by krusche** [comment]
@lovaiible Thanks! It now works great on mobile devices. However I found an issue on desktop devices. When I drag and drop an image in Safari in https://artemistest.ase.in.tum.de/#/quiz/534/practice, the "ghost image" is not visible any more. Could you have another look? On Chrome, it seems to work fine. I guess we need to test at least Firefox and Microsoft Edge as well.

@thilo-behnke @SiggZ I remember that you have Android devices, right? Could you please test the changes on the test server?

@jpbernius Could you please also test it on iOS mobile devices?

**Comment by krusche** [comment]
@lovaiible Ah and one more request. The background image on https://artemistest.ase.in.tum.de/#/quiz/534/practice is displayed way too large. When you are working on this anyway, can we somehow make sure that background images are not scaled larger than they originally are? This might be an easy fix. Down-scaling should be possible so that the image is not too large, but upscaling is not really necessary in my opinion. Maybe we can define the max-width?

File: src/main/webapp/app/components/util/secured-image.component.ts [hunk 8]
```diff
@@ -21,14 +21,21 @@ export const enum QuizEmitStatus {
 @Component({
     selector: 'jhi-secured-image',
     template: `
-        <img [attr.src]="dataUrl$ | async" />
+        <ng-template [ngIf]="this.dragAndDrop === undefined || this.dragAndDrop !== true">
```
Comment (thilo-behnke) [inline src/main/webapp/app/components/util/secured-image.component.ts]: Is it on purpose that it's just not `!this.dragAndDrop`?

**Comment by lovaiible** [comment]
@krusche I fixed the scaling of the background images. And the ghost images now work for Firefox and Safari again.

**Pull request merged**